### PR TITLE
Upgrade programs to Anchor to 0.22 + Release: Short circuit when withdraw amount is 0

### PR DESCRIPTION
**Pull request by gshohat**
-spl@22
vipers@2.0.1

### CODE REVIEW

File: programs/mint-proxy/src/lib.rs [hunk 1]
```diff
@@ -39,6 +39,13 @@ pub fn invoke_perform_mint<'a, 'b, 'c, 'info>(
     let mut acc_infos = ctx.to_account_infos();
     acc_infos.insert(0, mint_proxy_state);
     anchor_lang::solana_program::program::invoke_signed(&ix, &acc_infos, ctx.signer_seeds)
+        .map_err(Into::into)
+}
+
+impl From<ErrorCode> for ProgramError {
```
Comment (michaelhly) [inline programs/mint-proxy/src/lib.rs:45]: What is this for?

File: programs/mint-proxy/src/lib.rs [hunk 2]
```diff
@@ -124,13 +131,16 @@ pub mod mint_proxy {
...
+        pub fn minter_add(&self, ctx: Context<MinterAdd>, allowance: u64) -> Result<()> {
             let minter_info = &mut ctx.accounts.minter_info;
             minter_info.minter = ctx.accounts.minter.key();
             minter_info.allowance = allowance;
 
             let (_, nonce) = Pubkey::find_program_address(
-                &[b"anchor".as_ref(), minter_info.minter.key().as_ref()],
+                &[
+                    b"anchor".as_ref(),
+                    minter_info.minter.key().to_bytes().as_ref(),
```
Comment (michaelhly) [inline programs/mint-proxy/src/lib.rs:142]: I think you can just drop `.key()` here?

**Review by michaelhly — COMMENTED** [review]
Can you use the new anchor `bumps.get` method to generate the bump seed here:
https://github.com/saber-hq/saber-periphery/blob/65d056b2391778bdefd2e4d73b4fc08a4dcb3836/programs/mint-proxy/src/lib.rs#L139

File: programs/mint-proxy/src/lib.rs [hunk 3]
```diff
@@ -39,6 +39,13 @@ pub fn invoke_perform_mint<'a, 'b, 'c, 'info>(
     let mut acc_infos = ctx.to_account_infos();
     acc_infos.insert(0, mint_proxy_state);
     anchor_lang::solana_program::program::invoke_signed(&ix, &acc_infos, ctx.signer_seeds)
```
Comment (michaelhly) [inline programs/mint-proxy/src/lib.rs:41]: `anchor_lang::solana_program::program::invoke_signed(&ix, &acc_infos, ctx.signer_seeds)?`

not work?

File: programs/mint-proxy/src/lib.rs [hunk 4]
```diff
@@ -39,6 +39,13 @@ pub fn invoke_perform_mint<'a, 'b, 'c, 'info>(
...
+        .map_err(Into::into)
```
Comment (michaelhly) [inline programs/mint-proxy/src/lib.rs:42]:
```suggestion
        Ok(())
```

File: programs/mint-proxy/src/lib.rs (hunk 1 above)
Comment (gshohat) [reply programs/mint-proxy/src/lib.rs:45]: for the errorCode which is not implemented

**Pull request merged**